tests/andromede/test_ac_link.py

```diff
@@ -0,0 +1,385 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+import pytest
+
+from andromede.expression import param, var
+from andromede.expression.expression import literal, port_field
+from andromede.expression.indexing_structure import IndexingStructure
+from andromede.libs.standard import BALANCE_PORT_TYPE, DEMAND_MODEL, GENERATOR_MODEL
+from andromede.model import (
+    Constraint,
+    ModelPort,
+    PortField,
+    PortType,
+    float_parameter,
+    float_variable,
+    model,
+)
+from andromede.model.model import PortFieldDefinition, PortFieldId
+from andromede.simulation import OutputValues, TimeBlock, build_problem
+from andromede.study import (
+    ConstantData,
+    DataBase,
+    Network,
+    Node,
+    PortRef,
+    create_component,
+)
+
+"""
+Port transfers flow and voltage angle.
+"""
+AC_PORT = PortType(id="AC_LINK", fields=[PortField("angle"), PortField("flow")])
+
+"""
+Node has 2 ports: one for angle dependent connections,
+one for power-only connections.
+Should we relax constraints on ports compatibility to allow to have only one here?
+"""
+AC_NODE_MODEL = model(
+    id="AC_NODE",
+    variables=[float_variable("angle")],
+    ports=[
+        ModelPort(port_type=BALANCE_PORT_TYPE, port_name="injections"),
+        ModelPort(port_type=AC_PORT, port_name="links"),
+    ],
+    port_fields_definitions=[
+        PortFieldDefinition(
+            port_field=PortFieldId("links", "angle"), definition=var("angle")
+        )
+    ],
+    binding_constraints=[
+        Constraint(
+            name="Balance",
+            expression=port_field("injections", "flow").sum_connections()
+            + port_field("links", "flow").sum_connections()
+            == literal(0),
+        )
+    ],
+)
```

Comment:
To avoid infeasibilities, maybe add unsupplied energy and spilled energy varaibles

Reply:
For now I have not added this, since it's not useful in those first tests, but I agree it will be interesting to add it and add some related test cases.